### DISCUSSION TITLE
Possible to access $page.data in ParamMatcher?

### DISCUSSION THREAD

**Discussion by richarddavenport** [sveltejs/kit#8313] [Q&A]
I'm trying to access the `$page.data` in a `ParamMatcher` using `get(page)`. Internally sveltekit eventually invokes `getStores()`. From the docs:

> On the server, this must be called during component initialization.

`getStores()` throws an error (`Function called outside component initialization`) due to the fact that it's matching routes and doesn't appear to have started component initialization yet.

Is there another way to access `$page.data` in a `ParamMatcher`?

**Comment by dominikg** [comment]
No, param matchers are used to identify the route to load, page data is built using that route.  -> Chicken, meet Egg

**Reply by richarddavenport** [reply]
Yeah, I get it... I'm a little bummed and I know a lot of thought went into not having waterfalls. In my particular case I'm wanting data that is loaded in the root layout. Surely there must be a way to access this given it will be loaded in all routes?

**Reply by dominikg** [reply]
it is going to be loaded only after the matcher has already run. [matchers](https://kit.svelte.dev/docs/advanced-routing#matching) are simple `string=>boolean` functions for a reason, they run on both the server and the client and should be very lightweight / self explanatory.

What exactly are you trying to achieve? There are possibly other ways to get there, maybe[ layout groups](https://kit.svelte.dev/docs/advanced-routing#advanced-layouts) can help?

**Reply by Sher-V** [reply]
@dominikg I have the same problem. I want to access a feature flag in a matcher and depending on its state return **true** or **false**.
The reason I want to do that is because I have 2 solutions (monolith and microfrontend) for 1 page and I want to resolve them depending on a feature flag.

 Looks like it's not quite possible.

**Reply by dominikg** [reply]
> @dominikg I have the same problem. I want to access a feature flag in a matcher and depending on its state return **true** or **false**. The reason I want to do that is because I have 2 solutions (monolith and microfrontend) for 1 page and I want to resolve them depending on a feature flag.
> 
> Looks like it's not quite possible.

this depends on the way you implement feature flags. async reads won't work but sync reads eg from environment or with buildtime constants do.

please start a new discussion or ask in `#questions-svelte-and-kit` on https://svelte.dev/chat
